Convert byChroms.Rnw to byChroms.Rmd

Rohit-Satyam opened the pull request:
Asking for review from @Bioconductor/sweave2rmd @jwokaty. Below attached is the HTML output of the vignette.
[byChroms.zip](https://github.com/Bioconductor/geneplotter/files/10983147/byChroms.zip)

Rohit-Satyam committed:
Converted byChroms.Rnw file to Rmd

Rohit-Satyam committed:
Update DESCRIPTION with BiocStyle, knitr, VignetteBuilder

jwokaty commented:
The following checklist will be used to review your PR:

- [ ] the `.Rmd` file knits to `HTML`
- [ ] `R CMD build` runs without errors or timeouts
- [ ] the tarball from `R CMD build` contains the `HTML` (check with the
  following by substituting the package name and vignette name `tar ztf
  package_name.tar.gz | grep 'doc/vignette_name'`)
- [ ] the `.Rnw` file has been removed
- [ ] in the `DESCRIPTION`, `BiocStyle` and `knitr` are listed in `Suggests`
- [ ] in the `DESCRIPITON`, the line `VignetteBuilder: knitr` exists
- [ ] any added lines use the same spacing and indents as the existing
  document
- [ ] if agreed with the Maintainer, the contributor is in the author list
  in the `DESCRIPTION` file; refer to the contribution guide for acceptable
  formats
- [ ] if agreed with the Maintainer, the contributor is in the author list
  in the vignette's YAML; refer to the contribution guide for acceptable formats
- [ ] if this pull request includes a conversion from using the separate `Author`
  and `Maintainer` lines into the `Authors@R` vector, check that the
  `Authors@R` vector includes the maintainer as specified with `role='cre'`.
- [ ] if this pull request includes a conversion from using the separate `Author`
  and `Maintainer` lines into the `Authors@R` vector, the `Maintainer` line
  is completely removed.
- [ ] `HTML` document is representative of the `PDF` in content and
  in general the presentation
- [ ] Where the contributor was not able to preserve the content and presentation
  of the `PDF` is noted as a comment in the pull request
- [ ] the R Markdown file is representative of the Sweave document and follows
  best practices, such as replacing links to Bioconductor packages with calls
  to BiocStyle's `Biocpkg()`

BerylKanali commented on vignettes/byChroms.Rmd:
````diff
@@ -0,0 +1,91 @@
+chromosomes are similar and there is actual recombination going on
+between them. So, we will take the expedient measure of assigning each of them 
+to just one chromosome.
+```{r fixdata}
````
Add space between line 57 and 58. It makes the code look neat. It will also makes it corresponds to the Rnw code.

Rohit-Satyam replied:
Done

BerylKanali commented on vignettes/byChroms.Rmd:
````diff
@@ -0,0 +1,91 @@
+wh2[1]
+```
+So somehow `r length(wh2)` of the genes are mapped to two
+different chromosomes.  Based on (OMIM)[https://www.omim.org/] the these genes
````
When embedding links the syntax is [word] then (link). Fix it on the word OMIM to make it render correctly on the HTML file.

Rohit-Satyam replied:
Done

BerylKanali commented on vignettes/byChroms.Rmd:
```diff
@@ -0,0 +1,91 @@
+title: "geneplotter:Graphics related functions for Bioconductor"
+author:
+- name: Robert Gentleman
+  affiliation: "Biocore Team"
```
You can remove this affiliation.

Rohit-Satyam replied:
Done

Rohit-Satyam replied:
Done

BerylKanali commented:
- [x] the .Rmd file knits to HTML
- [x] R CMD build runs without errors or timeouts
- [ ]  the tarball from R CMD build contains the HTML (check with the
following by substituting the package name and vignette name tar ztf package_name.tar.gz | grep 'doc/vignette_name')
- [x] the .Rnw file has been removed
- [x] in the DESCRIPTION, BiocStyle and knitr are listed in Suggests
- [x] in the DESCRIPITON, the line VignetteBuilder: knitr exists
- [x] any added lines use the same spacing and indents as the existing
document
- [x] if agreed with the Maintainer, the contributor is in the author list
in the DESCRIPTION file; refer to the contribution guide for acceptable
formats
 - [x] if agreed with the Maintainer, the contributor is in the author list in the vignette's YAML; refer to the contribution guide for acceptable formats
- [x] if this pull request includes a conversion from using the separate Author and Maintainer lines into the Authors@R vector, check that theAuthors@R vector includes the maintainer as specified with role='cre'.
- [x] if this pull request includes a conversion from using the separate Author and Maintainer lines into the Authors@R vector, the Maintainer line is completely removed.
- [x] HTML document is representative of the PDF in content and in general the presentation
- [x] Where the contributor was not able to preserve the content and presentation of the PDF is noted as a comment in the pull request
 - [x] the R Markdown file is representative of the Sweave document and follows best practices, such as replacing links to Bioconductor packages with calls to BiocStyle's Biocpkg()

Rohit-Satyam committed:
Make changes in .Rmd file

BerylKanali commented on vignettes/byChroms.Rmd:
````diff
@@ -49,18 +48,21 @@ table(lens)
 length(chrs2)
 table(unlist(chrs2))
 ```
+
````
No need to add the extra space since there is space after the backticks. Remove line 65.

Rohit-Satyam replied:
Done

BerylKanali commented on vignettes/byChroms.Rmd:
````diff
@@ -77,15 +79,17 @@ splits <- split(strand, chrs2)
 length(splits)
 names(splits)
 ```
+
````
Remove this extra space as well.

Rohit-Satyam replied:
Done

BerylKanali commented on vignettes/byChroms.Rmd:
````diff
@@ -77,15 +79,17 @@ splits <- split(strand, chrs2)
 ```{r chrloc}
  newChrClass <- buildChromLocation("hu6800")
 ```
+
````
Remove this extra space as well.

Rohit-Satyam replied:
Done

Rohit-Satyam replied:
Should I open a new pull request?

BerylKanali replied:
No need for a new PR. All commits should show on this one.

BerylKanali reviewed (changes requested):
Hi @Rohit-Satyam 
I made some few additional comments, take a look at them and make the changes.

jwokaty commented:
@BerylKanali @Rohit-Satyam Have all changes been made and it is ready for me?

BerylKanali commented:
@jwokaty 
Not yet. I had asked @Rohit-Satyam to make some minor changes then it will be good to go.

Rohit-Satyam committed:
Update byChroms.Rmd

Removed spaces.

BerylKanali commented on vignettes/byChroms.Rmd:
```diff
@@ -0,0 +1,92 @@
+    BiocStyle::html_document:
+      toc_float: true
+    BiocStyle::pdf_document: default
+
```
Remove extra space that is line 25

Rohit-Satyam replied:
Done

BerylKanali commented on vignettes/byChroms.Rmd:
```diff
@@ -0,0 +1,92 @@
+    BiocStyle::pdf_document: default
+
+---
+
```
line 27 as well should be removed

Rohit-Satyam replied:
Done

BerylKanali reviewed (changes requested):
Hi @Rohit-Satyam 
Kindly go through your work one more time to ensure there are no extra spaces. After that let me know.

Rohit-Satyam commented:
Should I just use this PR link in to register contribution in Outrechy?

BerylKanali commented:
> Should I just use this PR link in to register contribution in Outrechy?

Yes

mcarlsn commented:
> Asking for review from @Bioconductor/sweave2rmd @jwokaty. Below attached is the HTML output of the vignette. [byChroms.zip](https://github.com/Bioconductor/geneplotter/files/10983147/byChroms.zip)

@Rohit-Satyam Thanks so much for your work on this conversion. We're in the process of organizing the Sweave2Rmd project board post-Outreachy contribution period ending and I wanted to follow-up with you on this vignette. Are you interested in completing the conversion? If not, no worries - we can pass it along to someone else - but I wanted to check with you first. Let us know!

Rohit-Satyam commented:
Hi @mcarlsn. I am sorry I am replying this two weeks later and this is because for some reason I am not getting github notifications on my gmails. I will resolve this issue. Besides, yes I would like to complete this package vignette conversion if it has not been assigned to someone else yet. Again apologies.

Edit: I saw your emails were in my spam folder. I sincerely apologize for the inconvenience. I thought since there was no notifications that this pull request might have been successfully merged.

Rohit-Satyam committed:
Removed extra spaces

Rohit-Satyam committed:
Adding visualize.Rmd

Rohit-Satyam commented:
I have also added `visualize.Rmd` as requested [here](https://github.com/Rohit-Satyam/geneplotter/blob/geneplotter-rmd/vignettes/visualize.Rmd). Kindly review @mcarlsn

Rohit-Satyam committed:
Edited Authors contribution section

jwokaty commented on vignettes/byChroms.Rmd:
```diff
@@ -0,0 +1,90 @@
+---
+title: "geneplotter:Graphics related functions for Bioconductor"
```
Change the title to `How to assemble a chromLocation object`.

Rohit-Satyam replied:
Done

jwokaty commented on vignettes/byChroms.Rmd:
```diff
@@ -0,0 +1,90 @@
+package: geneplotter
+      
+vignette: >
+ %VignetteIndexEntry{geneplotter:Graphics related functions for Bioconductor}
```
Generally, the `VignetteIndexEntry` is the same as the title. Please change to `How to assemble a chromLocation object`.

Rohit-Satyam replied:
Done

jwokaty commented on vignettes/byChroms.Rmd:
```diff
@@ -0,0 +1,90 @@
+  affiliation: "Vignette translation from Sweave to Rmarkdown / HTML"
+
+clean: false
+date: "`r Sys.Date()`"
```
For the date, use

    `r format(Sys.Date(), "%B %d, %Y")`

Rohit-Satyam replied:
Done. I believe you mean 
```
date: "`r format(Sys.time(), '%d %B, %Y')`"` 
```
coz the above recommendation leads to error

```
Error in yaml::yaml.load(..., eval.expr = TRUE) : 
  Scanner error: while scanning for the next token at line 8, column 31 found character that cannot start any token at line 8, column 31
```

**Edit 1:** Okay your recommended format worked this time. It might have been some unknown hiccup.

jwokaty commented on vignettes/byChroms.Rmd:
````diff
@@ -0,0 +1,90 @@
+ newChrClass <- buildChromLocation("hu6800")
+```
+
+And finally we can test it by calling `cPlot`.
+```{r cPlot, fig=TRUE}
````
Add a blank line between 86 and 87.

Rohit-Satyam replied:
Done

jwokaty commented on vignettes/byChroms.Rmd:
```diff
@@ -0,0 +1,90 @@
+
+output: 
+    BiocStyle::html_document:
+      toc_float: true
```
Remove the table of contents since there is only one header.

Rohit-Satyam replied:
Done

jwokaty commented on vignettes/byChroms.Rmd:
````diff
@@ -0,0 +1,90 @@
+  collapse = TRUE,eval = T)
+```
+
+# How to Assemble a chromLocation Object
````
Remove line 32. This should be the title.

Rohit-Satyam replied:
Done

jwokaty commented on vignettes/byChroms.Rmd:
````diff
@@ -0,0 +1,90 @@
+And finally we can test it by calling `cPlot`.
+```{r cPlot, fig=TRUE}
+  library(geneplotter)
+  cPlot(newChrClass)
````
This plot is old and it's no longer correct if we look at the version from Bioconductor 2.0: https://bioconductor.org/packages/2.0/bioc/vignettes/geneplotter/inst/doc/byChroms.pdf. We can kind of get closer to it by doing `cPlot(newChrClass, as.character(1:22))` but we also need X, M, Y, so we might have to ask for additional help to fix this plot.

Rohit-Satyam replied:
Hi Pardon my ignorance. I thought it was due to update in related packages and changes in the plot was expected. But I can offer additional help as well. The code can be modified as follows to generate the exact plot.

```
library(geneplotter)
## Reorder Chromosomes
newChrClass@chromLocs <- newChrClass@chromLocs[order(as.numeric(names(newChrClass@chromLocs)))]
newChrClass@chromInfo <- newChrClass@chromInfo[order(as.numeric(names(newChrClass@chromInfo)))]
cPlot(newChrClass,useChroms = as.character(c(names(splits),"X","Y","M")))
```

jwokaty replied:
Thanks for fixing the plot.

jwokaty commented on vignettes/byChroms.Rmd:
```diff
@@ -0,0 +1,90 @@
+ %VignetteDepends{annotate, hu6800.db}
+ %VignetteKeywords{chromosomes}
+ %VignettePackage{geneplotter}
+ %VignetteAuthor{Robert Gentleman}
```
Lines 13-19 should have 2 spaces. Currently there is only 1.

Rohit-Satyam replied:
Done. But I don't understand why introducing all these spaces are so important when the Vignette is knitting just fine?

jwokaty replied:
I tried to respond to a similar comment on slack, but I'll respond here. The reason why we're converting vignettes to .Rmd is that .Rmd files are easier to maintain, but just because something is in .Rmd format doesn't mean it's automatically readable so the requests to correct spacing, wrapping long lines, etc., are style guidelines that we've been picking up along the way as we refine the sweave2rmd process. Some of these suggestions come from core members, such as reformatting long lines. Because we also work with maintainers on their packages and their standards can be quite different, possibly stricter, our PRs need to be in the best shape possible. 

This style guide really needs to be added to the documentation. I also hope that in the future, we'll have a package do all or most of this work for us.

jwokaty commented on vignettes/byChroms.Rmd:
````diff
@@ -0,0 +1,90 @@
+chrs2 <- factor(chrs2)
+length(chrs2)
+table(unlist(chrs2))
+```
````
Add a blank space between the ends of code blocks and text.

Rohit-Satyam replied:
done

jwokaty commented on vignettes/byChroms.Rmd:
````diff
@@ -0,0 +1,90 @@
+length(splits)
+names(splits)
+```
+Now we have processed the data and are ready to construct a new
````
Add a blank space after code blocks and before text.

Rohit-Satyam replied:
Done.

jwokaty reviewed (changes requested):
Thanks for this PR. We generally keep conversions separate (1 PR to 1 conversion) and in this case, we already have a good PR for visualize.Rmd; however, it is being held because it has an old plot that requires fixing before we can merge it. Unfortunately,  I think it is best to remove the visualize files here.

In addition to the items, I've marked, there are two additional issues.

The first is that the plot at the bottom of the vignette has changed over time if you compare it with the vignette in Bioc 2.0--and this is not anything I would expect you to catch; I only guessed it might have a problem because the other vignette has a problem.  I made a suggestion below that might get us closer, but I think I might have to ask for help to update it to get the X, M, and Y parts unless you can figure it.

The second problem is that running `R CMD build` doesn't generate the html in the doc path, which is how the build system will generate it. After removing visualize, it might be easier to troubleshoot the issue.

Rohit-Satyam committed:
Removal Requested

Removal of this vignette requested in [PR](https://github.com/Bioconductor/geneplotter/pull/2)

Rohit-Satyam committed:
Minor revision in Vignette Rmd

As per [PR](https://github.com/Bioconductor/geneplotter/pull/2)

Rohit-Satyam commented:
The improved vignette is attached below.
[byChroms.zip](https://github.com/Bioconductor/geneplotter/files/11413139/byChroms.zip)

jwokaty reviewed (changes requested):
Can you add visualize.Rnw back in? 

I still cannot find the html in the doc path after building the vignette, which is how it will be generated by the build system. I am not sure where the problem is.

When I run `BiocCheck::BiocCheck` on the package, I get the following:

```
* Checking parsed R code in R directory, examples, vignettes...
    Found @ in vignettes/byChroms.Rmd
    * NOTE: Use accessors; don't access S4 class slots via '@' in
      examples/vignettes.
```
Can you address these issues?

jwokaty commented on vignettes/byChroms.Rmd:
```diff
@@ -0,0 +1,94 @@
+package: geneplotter
+      
+vignette: >
+  %VignetteIndexEntry{How to Assemble a chromLocation Object}
+  %VignetteEngine{knitr::rmarkdown}
+  %VignetteEncoding{UTF-8}  
+  %VignetteDepends{annotate, hu6800.db}
+  %VignetteKeywords{chromosomes}
+  %VignettePackage{geneplotter}
+  %VignetteAuthor{Robert Gentleman}
```
These are missing the `\`; for example

    %\VignetteIndexEntry{How to Assemble a chromLocation Object}

This is why we're not getting an error but the html isn't generated in the doc path of the tarball. Could I ask you to fix this here and in the other vignettes? Both GeneMeta and ShortRead have the same problem.

Rohit-Satyam replied:
Done

Rohit-Satyam replied:
I have made amendments in all the vignettes PR I have opened so far. The PRs are

1. https://github.com/Bioconductor/ShortRead/pull/14
2. https://github.com/Bioconductor/GenomicAlignments/pull/28
3. https://github.com/Bioconductor/GeneMeta/pull/2
4. https://github.com/Bioconductor/geneplotter/pull/2
5. https://github.com/lawremi/ggbio/pull/168
6. https://github.com/lgeistlinger/EnrichmentBrowser/pull/44

jwokaty replied:
Thank you. I've been checking these to see if you updated any. I will look at this PR first. Enrichmentbrowser needs to be reviewed by the maintainer who will merge the PR. (Maybe he needs a gentle reminder?)

Rohit-Satyam committed:
Update byChroms.Rmd

Added the missing `\` as suggested by reviewer in [issue](https://github.com/Bioconductor/geneplotter/pull/2#discussion_r1193077881)

Rohit-Satyam committed:
Adding back visualize.Rnw

Added the visualize.Rnw as per request by reviewer in [issue](https://github.com/Bioconductor/geneplotter/pull/2#pullrequestreview-1425461134)

jwokaty committed:
Merge branch 'devel' into geneplotter-rmd

jwokaty commented:
@Rohit-Satyam Thanks for your contribution! @BerylKanali Thanks for reviewing!
